### ISSUE TITLE
Hats off to all of you sir...

### ISSUE THREAD

**Issue by akkias** [michelson/Dante#38]
I was searching for an editor like medium, I did found some too but wasn't quite happy with them and thought of going with one of them and started coding with it but later one day I googled again for editor and tada I found you DANTE the best editor in the market... I love you.

but yeah I have one question how does upload image works :)

**Comment by thangngoc89** [comment]
Here is an example code :
```
 <script type="text/javascript">
    editor = new Dante.Editor(
      {
        el: "#editor",
        upload_url: "/images.json", //it expect an url string in response like /your/server/image.jpg or http://app.com/images/image.jpg
    
      }
    );
    editor.start()
  </script>
```

Dante will send a post request to /images.json , your server receive it and return a image link (like /your/server/image.jpg ).

**Comment by akkias** [comment]
Wow! sir that was so fast, and I am eagerly waiting for an image actions

**Comment by michelson** [comment]
Hi @akkias , thanks for your kind comment.
As @thangngoc89 said the upload_url attribute will set the path where you have your image uploader which will return the upload image path in response.
If you need more help on this just reopen this ticket.

best!

**Comment by thangngoc89** [comment]
@michelson is there any plan on the response mapping? Currently, our api send back a json data and we don't want to edit the api.

**Comment by michelson** [comment]
@thangngoc89 currently there are no plans to change that behavior.
do you have some ideas of how the response mapping would be ? patches welcome!

**Comment by thangngoc89** [comment]
@michelson here is my example : 
I assume my api return this data : 
```
{
'link' : 'http://your.server/image.jpg'
}
```

There will be a callback function like : 
```
<script type="text/javascript">
    editor = new Dante.Editor(
      {
        el: "#editor",
        upload_url: "/images.json", //it expect an url string in response like /your/server/image.jpg or http://app.com/images/image.jpg
        upload_callback: function(data)
        {
            return data.link;
        }
      }
    );
    editor.start()
  </script>
```

If there is no upload_callback, default behaviour will be triggered

**Comment by michelson** [comment]
looks good, maybe we could use the same upload_url to accepts a string or function

**Comment by thangngoc89** [comment]
yes, a lot of js package do it this way. Just pass the result into a closure (if there is any) and do the same job.

**Comment by michelson** [comment]
@thangngoc89 please check the branch `upload_url_mapping` and test it using `bundle exec rackup config.ru` I think this is what you looking, I'll merge this later.

**Comment by michelson** [comment]
@thangngoc89  the upload_callback option was released as 0.0.10 , thanks !
